skrub/_datetime_encoder.py

```diff
@@ -316,7 +344,19 @@ def transform(self, column):
             extracted = _get_dt_feature(column, feature).rename(f"{name}_{feature}")
             extracted = sbd.to_float32(extracted)
             all_extracted.append(extracted)
-        return sbd.make_dataframe_like(column, all_extracted)
+
+        _new_features = []
+        for _case, t in self._required_transformers.items():
+            if _case == "day":
+                _feat = _get_dt_feature(column, "day")
+                _feat = sbd.rename(_feat, "day")
+                _new_features.append(t.fit_transform(_feat))
```

Comment:
we're refitting the encoders every time we call transform. rather, we want to fit them during fit_transform and then call only `transform` not `fit_transform`